### PR TITLE
fix: initialization and iterables bugfixes

### PR DESCRIPTION
**Pull request by hjaraujof**
## Changes/solution

- Variable initialization fixes
- Async iterable pattern fixed in some services

### CODE REVIEW

File: src/services/appServiceWebApp/data.ts [hunk 1]
```diff
@@ -63,17 +63,15 @@ export default async ({
...
-                  webApp
-                const region = lowerCaseLocation(location)
-                return {
-                  ...rest,
-                  appServicePlanId: appService.id,
-                  region,
-                  resourceGroupId,
-                  Tags: tags || {},
-                }
+              const { location, extendedLocation, identity, tags, ...rest } =
```
Comment (tyler-dunkel) [inline src/services/appServiceWebApp/data.ts]: is there a case where webApp is undefined? Do we need something like
```
if (!webApp) { continue; }
```

File: src/services/dns/data.ts [hunk 2]
```diff
@@ -50,9 +50,7 @@ export default async ({
         await tryCatchWrapper(
           async () => {
             for await (const dnsZone of dnsZoneListIterable) {
-              if (dnsZone) {
-                dnsZones.push({ ...dnsZone, resourceGroupId: rgName })
-              }
+              dnsZones.push({ ...dnsZone, resourceGroupId: rgName })
```
Comment (tyler-dunkel) [inline src/services/dns/data.ts]: ++ here

File: src/services/functionApp/data.ts [hunk 3]
```diff
@@ -67,10 +67,7 @@ export default async ({
             const functionAppsIterable =
               client.webApps.listByResourceGroup(rgName)
             for await (const functionApp of functionAppsIterable) {
-              if (
-                functionApp &&
-                functionApp.kind.includes(apiSelectors.functionApp)
-              ) {
+              if (functionApp.kind.includes(apiSelectors.functionApp)) {
```
Comment (tyler-dunkel) [inline src/services/functionApp/data.ts]: will functionApp always exist? will `kind` always exist?

File: src/services/appServiceWebApp/data.ts (hunk 1 above)
Comment (hjaraujof) [reply src/services/appServiceWebApp/data.ts]: Yes, we need to ask if `webApp` has a truthy value. I'm missing a commit where I added the validation for all async iterables.

**Review by tyler-dunkel — APPROVED** [review]

**Pull request merged**